src/documents/RemoteModelAPI.md

```diff
@@ -0,0 +1,316 @@
+                <dd>
+                    The component.
+                </dd>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+### fetchImpl
+
+<table>
+    <tbody>
+        <tr>
+            <th>Description</th>
+            <td>
+                Must be supplied by an implementor to provide the concrete implementation for fetching data from the remote source. It must return a promise.
+            </td>
+        </tr>
+        <tr>
+            <th>Default</th>
+            <td><code>"fluid.notImplemented"</code></td>
+        </tr>
+    </tbody>
+</table>
+
+### write
+
+<table>
+    <tbody>
+        <tr>
+            <th>Description</th>
+            <td>
+                Initiates a write request. See the <a href="#write-workflow">Write Workflow</a> for more details.
+            </td>
+        </tr>
+        <tr>
+            <th>Default</th>
+            <td><code>"fluid.remoteModelComponent.write"</code></td>
+        </tr>
+        <tr>
+            <th>Parameters</th>
+            <td>
+                <dt>that</dt>
+                <dd>
+                    The component.
+                </dd>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+### writeImpl
+
+<table>
+    <tbody>
+        <tr>
+            <th>Description</th>
+            <td>
+                Must be supplied by an implementor to provide the concrete implementation for writing data to the remote source. It must return a promise that is resolved/rejected after the write operation has been handled by the remote end point.
+            </td>
+        </tr>
+        <tr>
+            <th>Default</th>
+            <td><code>"fluid.notImplemented"</code></td>
+        </tr>
+        <tr>
+            <th>Parameters</th>
+            <td>
+                <dt>payload</dt>
+                <dd>
+                    The data, from <code>that.model.local</code>, to send to the remote source.
+                </dd>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Fetch Workflow
+
+Executing the `fetch` invoker will add a fetch request and returns a promise for the result. Only one request can be in flight (processing) at a time. If a write request is in flight, the fetch will be queued. If a fetch request is already in queue/flight, the result of that request will be passed along to the current fetch request. When a fetch request is in flight , it will trigger the `fetchImpl` invoker to perform the actual request.
+
+Two synthetic events, `onFetch` and `afterFetch`, are fired during the processing of a fetch. `onFetch` can be used to perform any necessary actions before running `fetchImpl`. `afterFetch` can be used to perform any necessary actions after running `fetchImpl` (e.g. updating the model, unblocking the queue). If promises returned from `onFetch`, `afterFetch`, or `fetchImpl` are rejected, the `onFetchError` event will be fired.
+
+![A flow diagram depicting the Fetch workflow](images/remoteModel_fetch_diagram.svg)
```

Comment:
Should try to say something in words explaining the "rebasing" workflow in  fluid.remoteModelComponent.updateModelFromFetch